test/integration/inbox/filter_test.go

```diff
@@ -0,0 +1,740 @@
+		},
+		{
+			name: "Filter by search should match across title, description, and preview_message",
+			params: Params{
+				filter: &inbox.InboxFilterRequest{
+					Search: "signal",
+				},
+				expectedCount: 3,
+			},
+			setup: func(t *testing.T, params *Params, db dbbuilder.DBTX, logger interface{}) context.Context {
+				unitBuilder := unitbuilder.New(t, db)
+				userBuilder := userbuilder.New(t, db)
+				formBuilder := formbuilder.New(t, db)
+				inboxBuilder := inboxbuilder.New(t, db)
+
+				org := unitBuilder.Create(unit.UnitTypeOrganization, unitbuilder.WithName("filter-search-allfields-org"))
+				unitRow := unitBuilder.Create(unit.UnitTypeUnit, unitbuilder.WithOrgID(org.ID), unitbuilder.WithName("filter-search-allfields-unit"))
+				user := userBuilder.Create()
+
+				email := testdata.RandomEmail()
+				userBuilder.CreateEmail(user.ID, email)
+				unitBuilder.AddMember(unitRow.ID, email)
+
+				// One form with keyword in title
+				formTitle := formBuilder.Create(
+					formbuilder.WithUnitID(unitRow.ID),
+					formbuilder.WithLastEditor(user.ID),
+					formbuilder.WithTitle("signal update"),
+					formbuilder.WithDescription("weekly report"),
+				)
+				// One form with keyword at start of description (so preview picks it up if needed)
+				formDesc := formBuilder.Create(
+					formbuilder.WithUnitID(unitRow.ID),
+					formbuilder.WithLastEditor(user.ID),
+					formbuilder.WithTitle("metrics report"),
+					formbuilder.WithDescription("signal metrics and notes"),
+				)
+				// One form with keyword in preview_message
+				formPrev := formBuilder.Create(
+					formbuilder.WithUnitID(unitRow.ID),
+					formbuilder.WithLastEditor(user.ID),
+					formbuilder.WithTitle("release notes"),
+					formbuilder.WithDescription("hotfix v1.2.5"),
+					formbuilder.WithPreviewMessage("signal deployed to prod"),
+				)
+
+				msg1 := inboxBuilder.CreateMessage(inbox.ContentTypeForm, formTitle.ID, unitRow.ID)
+				msg2 := inboxBuilder.CreateMessage(inbox.ContentTypeForm, formDesc.ID, unitRow.ID)
+				msg3 := inboxBuilder.CreateMessage(inbox.ContentTypeForm, formPrev.ID, unitRow.ID)
+
+				inboxBuilder.CreateUserInboxMessage(user.ID, msg1.ID)
+				inboxBuilder.CreateUserInboxMessage(user.ID, msg2.ID)
+				inboxBuilder.CreateUserInboxMessage(user.ID, msg3.ID)
+
+				params.userID = user.ID
+
+				return context.Background()
+			},
+			validate: func(t *testing.T, params Params, db dbbuilder.DBTX, result []inbox.ListRow) {
+				require.Len(t, result, params.expectedCount)
+				for _, msg := range result {
+					// The search matches any of title, description (reflected in preview when no preview set), or preview_message
+					title := ""
+					if msg.Title != nil {
+						title = msg.Title.(string)
+					}
+					preview := ""
+					if msg.PreviewMessage != nil {
+						preview = msg.PreviewMessage.(string)
+					}
+					require.True(t, (title != "" && (title == "signal update" || title == "metrics report" || title == "release notes")) ||
```

Comment:
The brittleness in this section is the most severe among all cases, and it also raises readability concerns.

```go
preview == "signal metrics and notes"[:24]
```

This line duplicates and hardcodes the application’s business logic (when `preview_message` is empty, it takes the first 24 characters of description as the preview). If the preview logic changes even slightly (increasing the limit from 24 to 30 characters or adding "..." at the end) this test will immediately break, even though the search functionality itself remains perfectly correct.

Also, it only verifies that “each returned result is one of the three expected results,” but not that “the returned result set is exactly those three".